linetools/isgm/abscomponent.py

```diff
@@ -287,15 +287,16 @@ def from_dict(cls, idict, coord=None, skip_abslines=False, **kwargs):
                         slf.attrib[ak] = np.array(slf.attrib[ak])
 
         # Deprecated column (again)
-        if Ntup is not None:
+        if (Ntup is not None) and ('logN' not in idict['attrib'].keys()):
```

Comment:
@profxj @profxj While testing, I was trying to instantiate a COSHalos object (pyigm.cgm.cos_halos), and the following line crashes:
`if (Ntup is not None) and ('logN' not in idict['attrib'].keys()):
`
It appears that the dict getting fed to the abscomponent.from_dict() function doesn’t have an ‘attrib’ key.  (`KeyError: 'attrib'`).  However, the dicts loaded from the json files for our CASBaH CGM survey do actually contain this key.  Thus, I'm starting to wonder if the issue is coming from deprecated formats for the COS-Halos survey json?

Reply:
How about before creating the `'attrib'` key in the dictionary before feed it into `abscomponent.from_dict()`?

Reply:
Not sure I totally understand when this would happen...do you mean before the survey data is output to json or as it's being loaded in?

Reply:
If the input dictionary (`idict`) to be fed to `abscomponent.from_dict()` does not have a key named `'attrib'` I suggest trying to create it before passing it to the method. For instance,

`idict['attrib'] = attrib_dict`

where `attrib_dict` is a dictionary with the attributes for AbsComponent (including LogN).